server/src/stateManagement.mjs

```diff
@@ -697,22 +697,25 @@ function doesUserExist(users,userId){
 
   //iterating over list of users in state to ensure duplicate user/appId
   for(var key in users){
+    let parsedExistingUserId = parseUser(users[key])
+    let existingUser = parsedExistingUserId.user;
+    let existingAppId = parsedExistingUserId.appId;
     if (users[key].includes("~")){
-      if (user && users[key].split("~")[0]==user){
+      if (user && existingUser==user){
```

Comment:
This line is comparing two objects with equality, which is probably not what you want. Your variable names make distinguishing what's an object and what's an atomic field difficult. You might benefit from object dereferencing syntax (e.g., `const { existingUser, existingAppId, existingGroupId } = parseUser(users[key]);`).

Reply:
Actually, I'm realizing now that these aren't objects. But I stand by my comment: either use object dereferencing so as to not even have variable names for the structs/objects or name them so that they are obviously objects, e.g., objUser, objExistingUser or similar.